networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/topichandlers/Eth2TopicHandler.java

```diff
@@ -160,6 +170,10 @@ protected ValidationResult handleMessageProcessingError(
       final PreparedGossipMessage message, final Throwable err) {
     final ValidationResult response;
     if (ExceptionUtil.hasCause(err, DecodingException.class)) {
+      debugDataDumper.saveGossipMessageDecodingError(
+          getTopic(),
+          message.getArrivalTimestamp().map(UInt64::toString).orElse("unknown"),
+          message.getOriginalMessage());
```

Comment:
we need to think about this a little, this will be a wall of numerics that someone has to parse with potentially thousands and thousands of very large numbers.

Reply:
That is already being done just below with the P2PLogger, so it is not much different. If we are concerned with that, we may need to update the P2PLogger's data input too.

Reply:
Ah I see what you mean, fixing now.